apps/server/src/modules/board/saga/copy-room-boards.step.ts

```diff
@@ -58,26 +59,53 @@ export class CopyRoomBoardsStep extends SagaStep<'copyRoomBoards'> {
 		const sourceRoom = await this.roomService.getSingleRoom(sourceRoomId);
 		const targetRoom = await this.roomService.getSingleRoom(targetRoomId);
 
-		const sourceStorageLocationReference = { id: sourceRoom.schoolId, type: StorageLocation.SCHOOL };
-		const targetStorageLocationReference = { id: targetRoom.schoolId, type: StorageLocation.SCHOOL };
+		const copyStatuses: CopyStatus[] = [];
+		for (const board of boards) {
+			const copyStatus = await this.columnBoardService.copyColumnBoard({
+				originalColumnBoardId: board.id,
+				targetExternalReference: {
+					type: BoardExternalReferenceType.Room,
+					id: targetRoomId,
+				},
+				sourceStorageLocationReference: { id: sourceRoom.schoolId, type: StorageLocation.SCHOOL },
+				targetStorageLocationReference: { id: targetRoom.schoolId, type: StorageLocation.SCHOOL },
+				userId,
+				targetSchoolId: targetRoom.schoolId,
+				copyTitle: board.title,
+			});
+			copyStatuses.push(copyStatus);
+		}
 
-		const copyStatuses = await Promise.all(
-			boards.map((board) =>
-				this.columnBoardService.copyColumnBoard({
-					originalColumnBoardId: board.id,
-					targetExternalReference: {
-						type: BoardExternalReferenceType.Room,
-						id: targetRoomId,
-					},
-					sourceStorageLocationReference,
-					targetStorageLocationReference,
-					userId,
-					targetSchoolId: targetRoom.schoolId,
-					copyTitle: board.title,
-				})
-			)
-		);
+		const status: CopyStatus = {
```

Comment:
I don't like the construction of the CopyStatus here, because IMO it doesn't belong in the scope of this method. I suspect you put here because you wanted to reuse as much of `swapLinkedIdsInBoards` as possible. But since you changed it a bit anyway, you could also change it more, i.e. to `swapLinkedIdsInBoards(copyStatuses: CopyStatus[])`. I would also change the return type to `Promise<void>` because you don't do anything with the returned value.